### DISCUSSION TITLE
异步下载(setPathAsync)支持多线程与并发

### DISCUSSION THREAD

**Discussion by Hainess** [Tencent/libpag#2311] [Ideas] [answered]
1. path 为 url 时，已支持下载与缓存，但是发现多个pag下载是同步下载的；
2. 在列表展示多个PagImageView时，同步下载，会导致首次展示非常慢；

希望能支持多线程 并发数控制

**Comment by kevingpqi123** [comment, marked as answer]
关于 pag url 文件的下载，pag 这边只做了基础的下载功能，关于多线程和并发控制这个需求，由于人力有限，我们现在没有人力来完善这部分
大部分情况下 pag 文件都很小，下载速度会很快，如果是列表场景，且使用了较大的 pag 文件，如果你们的 app 中包含了多线程下载模块，下载这部分可以复用
